transformer/test_task.py

```diff
@@ -2,256 +2,140 @@
 
 import io
 import json
+from datetime import datetime
 from unittest.mock import MagicMock
 from unittest.mock import patch
 from urllib.parse import urlparse
 
 import pytest
 
 from transformer.request import Header
+from transformer import python as py
 from transformer.task import (
     Task,
     Request,
     HttpMethod,
     QueryPair,
     TIMEOUT,
     LocustRequest,
+    Task2,
 )
 
 
-class TestFromRequests:
-    def test_it_returns_a_task(self):
-        request = MagicMock()
-        request.timestamp = 1
-        second_request = MagicMock()
-        second_request.timestamp = 2
-        assert all(
-            isinstance(t, Task) for t in Task.from_requests([request, second_request])
-        )
-
-    @patch("builtins.open")
-    def test_it_doesnt_create_a_task_if_the_url_is_on_the_blacklist(self, mock_open):
-        mock_open.return_value = io.StringIO("amazon")
-        request = MagicMock()
-        request.url = MagicMock()
-        request.url.netloc = "www.amazon.com"
-        task = Task.from_requests([request])
-        assert len(list(task)) == 0
-
-    @patch("builtins.open")
-    def test_it_creates_a_task_if_the_path_not_host_is_on_the_blacklist(
-        self, mock_open
-    ):
-        mock_open.return_value = io.StringIO("search\namazon")
-        request = MagicMock()
-        request.url = urlparse("https://www.google.com/search?&q=amazon")
-        task = Task.from_requests([request])
-        assert len(list(task)) == 1
-
-
-class TestAsLocustAction:
-    def test_it_returns_an_error_given_an_unsupported_http_method(self):
-        a_request_with_an_unsupported_http_method = MagicMock()
-        task = Task("some_task", a_request_with_an_unsupported_http_method)
-        with pytest.raises(ValueError):
-            task.as_locust_action()
-
-    def test_it_returns_a_string(self):
-        a_request = MagicMock(spec_set=Request)
-        a_request.method = HttpMethod.GET
-        task = Task("some_task", a_request)
-        assert isinstance(task.as_locust_action(), str)
-
-    def test_it_returns_action_from_locust_request(self):
-        a_request = MagicMock(spec_set=Request)
-        a_request.method = HttpMethod.GET
-        locust_request = LocustRequest(
-            method=HttpMethod.GET, url=repr("http://locust-task"), headers={}
-        )
-        task = Task("some_task", request=a_request, locust_request=locust_request)
-        action = task.as_locust_action()
-        assert action.startswith("response = self.client.get(url='http://locust-task'")
-
-    def test_it_returns_task_using_get_given_a_get_http_method(self):
-        a_request = MagicMock(spec_set=Request)
-        a_request.method = HttpMethod.GET
-        task = Task("some_task", a_request)
-        action = task.as_locust_action()
-        assert action.startswith("response = self.client.get(")
-
-    def test_it_returns_a_task_using_post_given_a_post_http_method(self):
```

Comment:
It seems to me that - with all these tests deleted - there are now some cases that are left not covered with tests. One example is the `_parse_post_data` method. Please add more tests to cover existing code.

Reply:
Tests have been reworked, resolving for now.